server/api/impl/proxy/proxy.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package proxy
+
+import (
+	"encoding/json"
+	"net/http"
+
+	pluginapi "github.com/mattermost/mattermost-plugin-api"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-apps/server/api"
+	"github.com/mattermost/mattermost-plugin-apps/server/api/impl/upstream/upawslambda"
+	"github.com/mattermost/mattermost-plugin-apps/server/api/impl/upstream/uphttp"
+)
+
+type Proxy struct {
+	builtIn map[api.AppID]api.Upstream
+
+	mm    *pluginapi.Client
+	conf  api.Configurator
+	store api.Store
+}
+
+var _ api.Proxy = (*Proxy)(nil)
+
+func NewProxy(mm *pluginapi.Client, conf api.Configurator, store api.Store) *Proxy {
+	return &Proxy{nil, mm, conf, store}
+}
+
+func (p *Proxy) Call(debugSessionToken api.SessionToken, c *api.Call) *api.CallResponse {
+	app, err := p.store.LoadApp(c.Context.AppID)
+	if err != nil {
+		return api.NewErrorCallResponse(err)
+	}
+	up, err := p.upstreamForApp(app)
+	if err != nil {
+		return api.NewErrorCallResponse(err)
+	}
+
+	expander := p.newExpander(c.Context, p.mm, p.conf, p.store, debugSessionToken)
+	expander.App = app
+	cc, err := expander.Expand(c.Expand)
+	if err != nil {
+		return api.NewErrorCallResponse(err)
+	}
+	clone := *c
+	clone.Context = cc
+
+	return up.InvokeCall(&clone)
+}
+
+func (p *Proxy) Notify(cc *api.Context, subj api.Subject) error {
+	app, err := p.store.LoadApp(cc.AppID)
```

Comment:
Is `cc.AppID` ever defined for calls to this `Notify` function? It seems the `api.Context` is generated without an `AppID`, such as the call to [`api.NewUserContext(user)`](https://github.com/mattermost/mattermost-plugin-apps/blob/e887a213b6dab2fc904762b3e2a4d694ecafa1df/server/plugin.go#L106), which has no way of indicating an `AppID`.